src/formatter/Formatter.hx

```diff
@@ -182,6 +185,11 @@ enum FormatterInput {
 	#if (sys || nodejs)
 	FileInput(fileName:String);
 	#end
-	Code(code:String);
+	Code(code:String, origin:CodeOrigin);
 	Tokens(tokenList:Array<Token>, tokenTree:TokenTree, code:Bytes);
```

Comment:
`Tokens` needs an origin as well, since languageserver uses it to format documents using an existing tokentree structure. 
it currently maps to `<unknown.hx>` and which voids all attempts to exclude individual files

Reply:
As far as I can see `Tokens` are not used anywhere, although there is a case that handles them. Nonetheless it is not big deal to add origin for them as well. I don't know how the `fileName` should look for them. Is it the same like with `Code` (fileName: "path/to/file.ext") or should be in `<>`(fileName: "<path/to/file.ext>")

Reply:
we should treat it identical to `Code`. so instead of `<unknown.hx>` it would be `code snippet` and files use their actual filenames.

you can see `Tokens` used in [languageserver](https://github.com/vshaxe/haxe-language-server/blob/62657538ba363efb2b758e67c2b034540506e256/src/haxeLanguageServer/features/haxe/DocumentFormattingFeature.hx#L56)

Reply:
I see... Thanks. 
If I change the Tokens adding origin, wouldn't that break the languageserver then?